test/e2e/receive_test.go

```diff
@@ -496,9 +496,9 @@ func TestReceive(t *testing.T) {
 
 		h := receive.HashringConfig{
 			Endpoints: []string{
-				r1.InternalEndpoint("grpc"),
-				r2.InternalEndpoint("grpc"),
-				r3.InternalEndpoint("grpc"),
+				r1.Future().InternalEndpoint("grpc"),
```

Comment:
I did not design it to do Future() all the time, so maybe r1 = r1.Future would make more sense, but not a big deal